File: app/src/main/java/socioverse/tifor/Activity/CallNameActivity.java
package socioverse.tifor.Activity;

import android.app.Application;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.zegocloud.uikit.prebuilt.call.config.ZegoNotificationConfig;
import com.zegocloud.uikit.prebuilt.call.invite.ZegoUIKitPrebuiltCallInvitationConfig;
import com.zegocloud.uikit.prebuilt.call.invite.ZegoUIKitPrebuiltCallInvitationService;

import socioverse.tifor.databinding.ActivityCallNameBinding;
import socioverse.tifor.utlities.PreferenceManager;

public class CallNameActivity extends BaseActivity2 {

    private ActivityCallNameBinding binding;
    private PreferenceManager preferenceManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityCallNameBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        preferenceManager = new PreferenceManager(getApplicationContext());

        binding.send.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                String userId = binding.userId.getText().toString().trim();
                if (userId.isEmpty()) {

                } else {

                    try {
                        startService(userId);
                        Intent intent = new Intent(getApplicationContext(), CallActivity.class);
                        intent.putExtra("userId", userId);
                        startActivity(intent);
                    } catch (Exception e) {

                    }

                }


            }
        });

    }

    private void startService(String userId) {
        try {

            Application application = getApplication(); // Android's application context
            long appID = 900851183;   // yourAppID
            String appSign = "e2dbd39a96fcba5c14649d787451893020a522a12891490ac92bd90e2a62ad3f";  // yourAppSign
            String userID = userId; // yourUserID, userID should only contain numbers, English characters, and '_'.
            String userName = userId;  // yourUserName

            ZegoUIKitPrebuiltCallInvitationConfig callInvitationConfig = new ZegoUIKitPrebuiltCallInvitationConfig();
            callInvitationConfig.notifyWhenAppRunningInBackgroundOrQuit = true;
            ZegoNotificationConfig notificationConfig = new ZegoNotificationConfig();
            notificationConfig.sound = "zego_uikit_sound_call";
            notificationConfig.channelID = "CallInvitation";
            notificationConfig.channelName = "CallInvitation";
            ZegoUIKitPrebuiltCallInvitationService.init(getApplication(), appID, appSign, userID, userName, callInvitationConfig);
        } catch (Exception e) {

        }

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        try {
            ZegoUIKitPrebuiltCallInvitationService.unInit();
        } catch (Exception e) {

        }
    }
}

File: app/src/main/java/socioverse/tifor/Activity/CallNameActivity.kt
package socioverse.tifor.Activity

import android.app.Application
import android.content.Intent
import android.os.Bundle
import android.view.View
import com.zegocloud.uikit.prebuilt.call.config.ZegoNotificationConfig
import com.zegocloud.uikit.prebuilt.call.invite.ZegoUIKitPrebuiltCallInvitationConfig
import com.zegocloud.uikit.prebuilt.call.invite.ZegoUIKitPrebuiltCallInvitationService
import socioverse.tifor.databinding.ActivityCallNameBinding
import socioverse.tifor.utlities.PreferenceManager

class CallNameActivity : BaseActivity2() {
    private lateinit var binding: ActivityCallNameBinding
    private lateinit var preferenceManager: PreferenceManager
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityCallNameBinding.inflate(layoutInflater)
        setContentView(binding.root)

        preferenceManager = PreferenceManager(applicationContext)


        binding.send.setOnClickListener(object : View.OnClickListener {
            override fun onClick(view: View) {
                val userId: String = binding.userId.text.toString().trim()
                if (userId.isEmpty()) {
                } else {
                    try {
                        startService(userId)
                        val intent: Intent =
                            Intent(applicationContext, CallActivity::class.java)
                        intent.putExtra("userId", userId)
                        startActivity(intent)
                    } catch (e: Exception) {
                    }
                }
            }
        })
    }

    private fun startService(userId: String) {
        try {
            val application: Application = application // Android's application context
            val appID: Long = 900851183 // yourAppID
            val appSign: String =
                "e2dbd39a96fcba5c14649d787451893020a522a12891490ac92bd90e2a62ad3f" // yourAppSign
            val userID: String =
                userId // yourUserID, userID should only contain numbers, English characters, and '_'.
            val userName: String = userId // yourUserName
            val callInvitationConfig: ZegoUIKitPrebuiltCallInvitationConfig =
                ZegoUIKitPrebuiltCallInvitationConfig()
            callInvitationConfig.notifyWhenAppRunningInBackgroundOrQuit = true
            val notificationConfig: ZegoNotificationConfig = ZegoNotificationConfig()
            notificationConfig.sound = "zego_uikit_sound_call"
            notificationConfig.channelID = "CallInvitation"
            notificationConfig.channelName = "CallInvitation"
            ZegoUIKitPrebuiltCallInvitationService.init(
                getApplication(), appID, appSign, userID, userName, callInvitationConfig
            )
        } catch (e: Exception) {
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        try {
            ZegoUIKitPrebuiltCallInvitationService.unInit()
        } catch (e: Exception) {
        }
    }
}